### PR TITLE
[REFACTOR] use bolt11 library

### PR DESCRIPTION
**Pull request by dni**
refactored to use https://pypi.org/project/bolt11/

### CODE REVIEW

File: cashu/mint/ledger.py [hunk 1]
```diff
@@ -927,7 +927,8 @@ async def melt(
             logger.trace("verified proofs")
 
             total_provided = sum_proofs(proofs)
-            invoice_obj = bolt11.decode(invoice)
+            invoice_obj = decode(invoice)
```
Comment (AngusP) [inline cashu/mint/ledger.py:930]: Minor: would prefer `import bolt11` then `bolt11.decode(invoice)`, here it's a bit of a "huh which decode" when reading (e.g. I'd think of the stdlib `bytes.decode` first)

**Review by AngusP — APPROVED** [review]
LGTM (I've not tested though)

File: cashu/mint/ledger.py (hunk 1 above)
Comment (dni) [reply cashu/mint/ledger.py:930]: is `from bolt11.decode import decode as bolt11_decode` ok?

Comment (AngusP) [reply cashu/mint/ledger.py]: sure

**Review by callebtc — APPROVED** [review]

**Pull request closed without merging**